### PR TITLE
AB#445 AB#466 Textbox type for priority question has no options to add values

### PR DESCRIPTION
**Pull request by franciscoghersi**

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts [hunk 1]
```diff
@@ -179,25 +181,41 @@ export class QuestionsCreateComponent implements OnDestroy, OnInit {
...
+            questionOption.minimumRelative = o.minRelativeFormControl;
+            questionOption.maximumRelative = o.maxRelativeFormControl;
+            questionOption.weight = o.optionWeight;
+            options = [...options, questionOption];
+          }
+          if (!this.optionsRange(options)) {
             this.modal.error({
               nzTitle: 'Warning',
-              nzContent: 'The weight of each option must sum a total of 100',
+              nzContent: 'The range does not cover the entire value.',
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts:207]: Is this text shown to User? consider using translation.

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts [hunk 2]
```diff
@@ -179,25 +181,41 @@ export class QuestionsCreateComponent implements OnDestroy, OnInit {
...
           }
         }
+        questionSavedItem.options = options;
 
+        if (this.optionsWeight(questionSavedItem.options) !== true) {
+          this.modal.error({
+            nzTitle: 'Warning',
+            nzContent: 'The weight of each option must sum a total of 100',
+          });
+        }
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts:218]: Method addQuestion() has too many line codes, and code repeated (Edit mode and New item mode are very similar). Pease refactor in private methods. Ask @sculebras for tips.

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.html [hunk 3]
```diff
@@ -121,7 +121,73 @@ <h4 class="section-title">{{ 'Admin.PriorityQuestion.Create.Title' | translate }
               </nz-select>
             </nz-form-control>
           </nz-form-item>
-          <nz-form-item>
+
+          <!-- textbox -->
+          <nz-form-item *ngIf="controlTypeId == 1">
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.html:126]: Match this with an enum (see .ts file comparission by type text) and use === to compare

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.html [hunk 4]
```diff
@@ -121,7 +121,73 @@ <h4 class="section-title">{{ 'Admin.PriorityQuestion.Create.Title' | translate }
...
+              <nz-form-control>
+                <button nz-button nzType="dashed" class="add-button" (click)="addField($event)">
+                  <i nz-icon nzType="plus"></i>
+                  Add field
+                </button>
+              </nz-form-control>
+            </nz-form-item>
+          </nz-form-item>
+
+          <nz-form-item *ngIf="controlTypeId > 1">
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.html:190]: Match this with an enum (see .ts file comparission by type text) and use !== to compare

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts [hunk 5]
```diff
@@ -179,25 +181,41 @@ export class QuestionsCreateComponent implements OnDestroy, OnInit {
         questionSavedItem.weight = this.questionItemFormGroup.controls.weightFormControl.value;
         questionSavedItem.defaultValue = this.questionItemFormGroup.controls.defaultValueFormControl.value;
 
+        this.optionsArray.removeAt(this.optionsArray.length);
         if (questionSavedItem.controlType.name !== 'Textbox') {
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts:185]: Move this to an enum

File: src/DonateTo.Infrastructure/Data/Migrations/20200727195825_UpdateQuestionMinMaxValue.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace DonateTo.Infrastructure.Migrations
+{
+    public partial class UpdateQuestionMinMaxValue : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (guillemurano) [inline src/DonateTo.Infrastructure/Data/Migrations/20200727195825_UpdateQuestionMinMaxValue.cs:7]: Being a percentage I would reccomend using a decimal number for these values. If any of them can be empty change the nullable to true which will make the default value not needed.

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.html [hunk 7]
```diff
@@ -184,7 +187,7 @@ <h4 class="section-title">{{ 'Admin.PriorityQuestion.Create.Title' | translate }
             </nz-form-item>
           </nz-form-item>
 
-          <nz-form-item *ngIf="controlTypeId > 1">
+          <nz-form-item *ngIf="controlTypeId !== controlTypeEnum.Textbox && controlTypeId > 0">
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.html:190]: No need to compare greater than zero here as it's already initialized as zero

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts [hunk 8]
```diff
@@ -157,23 +167,9 @@ export class QuestionsCreateComponent implements OnDestroy, OnInit {
...
-    this.defaultValue = '';
-    this.controlTypeId = 0;
-    this.questionId = 0;
-    this.isEdit = false;
-    this.optionsArray = new FormArray([]);
-    this.addField();
-    this.addField();
-  }
-
   addQuestion() {
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts:170]: addQuestion method has too many lines and repeated code found. Edit mode and new item mode seems to be similar.
Please refactor. ask @sculebras for tips

Comment (franciscoghersi) [reply src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.ts:170]: Ok, i will refact that

**Comment by guillemurano** [comment]
@franciscoghersi please check conflicts

**Pull request merged**